src/main/scala/cromwell/binding/values/WdlFileLike.scala

```diff
@@ -0,0 +1,10 @@
+package cromwell.binding.values
+
+import cromwell.binding.types.WdlFileType
+
+/**
+ * WDL value which identifies the location of a File-like object.
+ */
+trait WdlFileLike extends WdlPrimitive {
+  val wdlType = WdlFileType
```

Comment:
If I understand correctly, the implementation of the engine functions makes the decision.  The thing that's confusing to me is that there's WdlGcsObject which extends WdlFileLike... but WdlFileLike has a type of WdlFile.  So indirectly we're saying `WdlGcsObject` is a `WdlFile`.  I guess my expectation was that `WdlGcsObject` would infer `WdlUri` instead of `WdlFile`.

I think I'd get more out of this conversation if we just talked about it briefly on Monday, though because I feel like I may just be misinterpreting something.  I also need to fully digest the rest of this code.